### ISSUE TITLE
DNS container segmentation violation

### ISSUE THREAD

**Issue by sapakt** [rancher/rancher#10223]
**Rancher versions:**
rancher/server: 1.6.7
rancher/agent: 1.2.5
Cattle 0.183.5

**Infrastructure Stack versions:**
healthcheck: 0.3.0
ipsec: 0.1.3
network-services: 0.2.5
scheduler: 0.6.3
kubernetes (if applicable):

**Docker version: (`docker version`,`docker info` preferred)**
Client:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:38:45 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.6
 API version:  1.24
 Go version:   go1.6.4
 Git commit:   78d1802
 Built:        Tue Jan 10 20:38:45 2017
 OS/Arch:      linux/amd64
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
NAME="Ubuntu"
VERSION="16.04.2 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.2 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial
**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
AWS
**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
HA rancher, external DB
**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**

**Results:**

Occasionally some of our containers are unable to resolve DNS records. If we check rancher/dns logs on affected host, there is:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x5112e6]

goroutine 242269711 [running]:
panic(0x7c5100, 0xc420016060)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/rancher/rancher-dns/vendor/github.com/skynetservices/skydns/cache.(*Cache).Search(0x0, 0xc4205f4720, 0x2b, 0xbf0000, 0xc4205f4720, 0x2b, 0xc, 0xbf7e40)
        /go/src/github.com/rancher/rancher-dns/vendor/github.com/skynetservices/skydns/cache/cache.go:109 +0x26
github.com/rancher/rancher-dns/vendor/github.com/skynetservices/skydns/cache.(*Cache).Hit(0x0, 0xc4202360c0, 0x15, 0x1001c, 0xc4abc10000, 0xc4206a5a68)
        /go/src/github.com/rancher/rancher-dns/vendor/github.com/skynetservices/skydns/cache/hit.go:17 +0x97
main.clientSpecificCacheHit(0xc420236120, 0xc, 0xc420536090, 0x1)
        /go/src/github.com/rancher/rancher-dns/cache.go:37 +0xa3
main.route(0xbc28c0, 0xc420210070, 0xc420536090)
        /go/src/github.com/rancher/rancher-dns/main.go:264 +0x5df
github.com/rancher/rancher-dns/vendor/github.com/miekg/dns.HandlerFunc.ServeDNS(0x852138, 0xbc28c0, 0xc420210070, 0xc420536090)
        /go/src/github.com/rancher/rancher-dns/vendor/github.com/miekg/dns/server.go:82 +0x44
github.com/rancher/rancher-dns/vendor/github.com/miekg/dns.(*ServeMux).ServeDNS(0xc42007a850, 0xbc28c0, 0xc420210070, 0xc420536090)
        /go/src/github.com/rancher/rancher-dns/vendor/github.com/miekg/dns/server.go:186 +0x61
github.com/rancher/rancher-dns/vendor/github.com/miekg/dns.(*Server).serve(0xc4201f6000, 0xbbe080, 0xc42007d470, 0xbbb5c0, 0xc42007a850, 0xc4204d0000, 0x26, 0x200, 0xc42002c030, 0xc420236060, ...)
        /go/src/github.com/rancher/rancher-dns/vendor/github.com/miekg/dns/server.go:535 +0x4e6
created by github.com/rancher/rancher-dns/vendor/github.com/miekg/dns.(*Server).serveUDP
        /go/src/github.com/rancher/rancher-dns/vendor/github.com/miekg/dns/server.go:489 +0x2c8
time="2017-10-24T13:22:07Z" level=info msg="Starting rancher-dns v0.15.1"
```

Usually under one minute rancher-dns service is started and problem is resolved. Is there something we can do to resolve this issue?

**Comment by superseb** [comment]
There is a nil pointer fix in network-services 0.2.6, available in Rancher v1.6.8+. Can you try this and see this solves your case?

**Comment by sapakt** [comment]
We will need to plan the upgrade. Thank you for the information.
